selftests/functional/test_requirements.py

```diff
@@ -0,0 +1,143 @@
+        """
+'''
+
+SINGLE_FAIL_CHECK = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class FailTest(Test):
+
+    def test_check(self):
+        """
+        :avocado: requirement={"type": "package", "name": "-foo-bar-", "action": "check"}
+        """
+'''
+
+MULTIPLE_SUCCESS = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class SuccessTest(Test):
+
+    def test_a(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_b(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_c(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+'''
+
+MULTIPLE_FAIL = '''#!/usr/bin/env python3
+
+from avocado import Test
+
+
+class FailTest(Test):
+
+    def test_a(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        :avocado: requirement={"type": "package", "name": "-foo-bar-"}
+        """
+    def test_b(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        """
+    def test_c(self):
+        """
+        :avocado: requirement={"type": "package", "name": "hello"}
+        :avocado: requirement={"type": "package", "name": "-foo-bar-"}
+        """
+'''
+
+
+class BasicTest(TestCaseTmpDir):
+
+    status_server = '127.0.0.1:%u' % find_free_port()
+    command = ('%s run --test-runner=nrunner'
+               ' --nrunner-status-server-uri %s'
+               ' --nrunner-status-server-listen %s %s')
+
+    def test_single_success(self):
+        with script.Script(os.path.join(self.tmpdir.name,
+                                        'test_single_success.py'),
+                           SINGLE_SUCCESS_CHECK) as test:
+            command = self.command % (AVOCADO, self.status_server,
+                                      self.status_server, test.path)
+            result = process.run(command, ignore_status=True)
+            self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
+            self.assertIn('PASS 1', result.stdout_text,)
+            self.assertNotIn('bash', result.stdout_text,)
+
+    def test_single_fail(self):
+        with script.Script(os.path.join(self.tmpdir.name,
+                                        'test_single_fail.py'),
+                           SINGLE_FAIL_CHECK) as test:
+            command = self.command % (AVOCADO, self.status_server,
+                                      self.status_server, test.path)
+            result = process.run(command, ignore_status=True)
+            self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
+            self.assertIn('PASS 0', result.stdout_text,)
+            self.assertIn('SKIP 1', result.stdout_text,)
+            self.assertNotIn('-foo-bar-', result.stdout_text,)
+
+    @unittest.skipIf(not os.getenv('CI') and os.getenv('CI') not in ['true'],
```

Comment:
Can you better explain what a "CI" has, that another (non-CI) environment doesn't have?

As much as possible (and I recognize it's not always possible) we should try to determine the execution of a test based on what it needs (superuser? capability? a lot of time?) rather then based on a specific location.

Reply:
Yep, this was a leftover from the time I was testing the test :) and I forgot to update for a better description.

The idea here is "hey, I'll install or remove packages to your system and I may screw up", that is why I preferred to use an environment instead of checking for local capabilities/authorities.

Reply:
Then maybe let's add a "message" argument to the `skipIf`.  Also, it may be worth to add a custom decorator to `selftests/__init__.py`, such as `skipUnlessOnCI`, that can account for all CI types we know/use (not sure if they all set the `CI` env var).

Reply:
I checked the main CIs we currently use (not necessary on Avocado Project) and `CI=true` is available on Travis, GitHub Actions, and Gitlab CI. For now, I think it is enough to improve the message.